### ISSUE TITLE
Very High usage of RAM.

### ISSUE THREAD

**Issue by Azathoth-11** [gh0stzk/dotfiles#49]
When I first boot up in bspwm its shows about 700-800mb of RAM Usage. But when I open chrome or firefox it jumps for 1gb to 4gb. Then even if I close the appllcation its somehow keeps on increasing and finally it freezes.

Any solutions or either I am doing something wrong?

**Comment by gh0stzk** [comment]
That is very strange because it is only BSPWM

How do you log in, with a login manager like lightdm or ssddm, with initrc, startx???

What services do you have enabled to start with the system?

In addition to the programs that start automatically in my dotfiles, like polkit for example, have you added any more?

You have to take a look at the free -m command and check how your ram memory is being used.

Finally use htop or top, to see what program, service is growing like crazy.

I test my dotfiles on 3 machines, one with 16 gb ram, another with 8 and the last one with 3 gb, which is a 2011 machine and I have not had this problem.

However, some time ago another person wrote me with the same problem as you. I don't think it's bspwm + the dotfiles, it must be something very specific in the configuration, the services enabled, etc...

Initially you can start with htop and free -m to start to get an idea.

**Comment by wuhanhui** [comment]
I meet the some problem with high mem use. But if I check with htop I can not find high mem application

**Comment by wuhanhui** [comment]
I can reproduce it by opening Chrome and making this Chrome node to be floating. Then I change the node size by right holding the mouse. Then the mem is increasing.

**Comment by gh0stzk** [comment]
Maybe you can try disabling picom, comment the line from bspwmrc at bottom where picom is launched.
The one my dotfiles use is the official one from arch repos. 
Maybe trying installing bspwm-git and picom-git from AUR, just to check if issue is gone.
For last, i dont know what is the specs of your machine, maybe change to legacy-backends in picom.
I will play with picom options next days hopefully trying to find or reproduce the issue. In my low end and high end machines works fine, so well.. i will play with some options in picom

**Comment by gh0stzk** [comment]
i search a little and found some post in reddit with this issue, like this https://www.reddit.com/r/archlinux/comments/128sa6f/new_arch_vm_installs_with_bspwm_consuming_all_ram/

i search in bspwm github issues and even i found your comment 1 hour ago, https://github.com/baskerville/bspwm/issues/1422#issuecomment-1501040893

try with bspwm-git is some commits ahead from official repos. or compile it from source.
Is a bspwm issue

**Comment by wuhanhui** [comment]
Thanks for your suggestion. I try to install bspwm-git and comment all the picom in the config file. But the problem is still there.

**Comment by gh0stzk** [comment]
I just wrote to one of the maintainers hopefully can help with this.

**Comment by BenedettiLucca** [comment]
Same problem here, htop says it's lightdm that's eating all the ram
![ShoT_2023-04-14-02-20-26_3360x1080](https://user-images.githubusercontent.com/108427035/232113730-6eb01719-5111-4110-b327-7852b8459f3a.png)

**Comment by BenedettiLucca** [comment]
strangely fixed after installing `plymouth` and `xorg-server-xephyr`

**Comment by gh0stzk** [comment]
> strangely fixed after installing `plymouth` and `xorg-server-xephyr`

Yes, but I don't think that this behavior had anything to do with the dotfiles themselves. Good that you solved it

**Comment by gh0stzk** [comment]
For everyone else, it could be a problem with the picom configuration.

Try this:

In the picom.conf file located at ~/.config/bspwm/rices/rice name/picom.conf

Look for line 99:

backend="xrender";

Comment it and uncomment the line above, so that it looks like this:

backend="glx";
#backend = "xrender";

My dotfiles use the xrender backend by default, but we all have different hardware, different configuration, etc etc, some are fine with the default xrender, but others the glx backend.

You can even try the new picom backend called "egl"
backend="glx";

You need to test your setup with the different backends, dotfiles are only a start point, you need to adapt it to your hardware, needs an taste.

**Comment by wuhanhui** [comment]
Yes, It works for you when I change it to glx. Thanks a lot.

**Comment by gh0stzk** [comment]
> Yes, It works for you when I change it to glx. Thanks a lot.

Good!!

**Comment by gh0stzk** [comment]
Also i just did a wiki page for this

[https://github.com/gh0stzk/dotfiles/wiki/High-RAM-CPU-usage%3F%3F%3F](https://github.com/gh0stzk/dotfiles/wiki/High-RAM-CPU-usage%3F%3F%3F)

**Comment by Azathoth-11** [comment]
Its really working without any issue now. Thanks for the dotfiles.
